File: app/src/main/java/com/icerockdev/babenko/ui/images/ImagesView.java
package com.icerockdev.babenko.ui.images;

import com.icerockdev.babenko.model.entities.ImageItem;
import com.icerockdev.babenko.ui.base.views.ProgressBaseView;

import java.util.List;

/**
 * Created by Roman Babenko on 10/05/17.
 */

public interface ImagesView extends ProgressBaseView {
    void showErrorDialog();

    void showListIsEmptyError();

    void showImagesList(List<ImageItem> imageItems);
}

File: app/src/main/java/com/icerockdev/babenko/ui/images/ImagesView.kt
package com.icerockdev.babenko.ui.images

import com.icerockdev.babenko.model.entities.ImageItem
import com.icerockdev.babenko.ui.base.views.ProgressBaseView

/**
 * Created by Roman Babenko on 10/05/17.
 */

interface ImagesView : ProgressBaseView {
    fun showErrorDialog()

    fun showListIsEmptyError()

    fun showImagesList(imageItems: List<ImageItem>)
}
